src/main/java/org/spongepowered/api/world/border/WorldBorder.java

```diff
@@ -0,0 +1,234 @@
+ *
+ * <p>It can gradually grow or shrink to a radius over a period of time. A
+ * warning is displayed when a contracting world border will reach the player in
+ * a certain amount of time, or when the player is a certain number of blocks
+ * away.</p>
+ *
+ * <p>In Minecraft, a warning is displayed in the form of a reddish tint.</p>
+ */
+public interface WorldBorder {
+
+    /**
+     * Creates a new {@link Builder} instance for making {@link WorldBorder}s.
+     *
+     * @return The builder
+     */
+    static Builder builder() {
+        return Sponge.game().builderProvider().provide(Builder.class);
+    }
+
+    /**
+     * Creates a new {@link Builder} instance for making {@link WorldBorder}s,
+     * pre-populated with the values of this border.
+     *
+     * @return The builder
+     */
+    default Builder toBuilder() {
+        return WorldBorder.builder().from(this);
+    }
+
+    /**
+     * Gets the center of the world border.
+     *
+     * <p>The x coordinated can be retrieved through {@link Vector2d#x()}, while
+     * the z coordinated can be retrieved through {@link Vector2d#y()}.</p>
+     *
+     * @return The center
+     */
+    Vector2d center();
+
+    /**
+     * Gets the current diameter of this world border. If this is an immutable
+     * border, this is the initial diameter.
+     *
+     * <p>The returned diameter applies to the x and z axis. The world border
+     * extends over the entire y-axis.</p>
+     *
+     * @return The size
+     */
+    double diameter();
+
+    /**
+     * Gets the target diameter the world border is expanding or contracting to.
+     *
+     * <p>This will return the same value as {@link #diameter()} unless
+     * {@link #timeUntilTargetDiameter()} is greater than 0.</p>
+     *
+     * @return The diameter being changed to
+     */
+    double targetDiameter();
+
+    /**
+     * Gets the time remaining until the world border stops expanding or
+     * contracting to the {@link #targetDiameter()}.
+     *
+     * <p>If {@link #diameter()} and {@link #targetDiameter()} are the same,
+     * this will be {@link Duration#ZERO}.</p>
+     *
+     * @return The time remaining
+     */
+    Duration timeUntilTargetDiameter();
+
+    /**
+     * Gets the distance a player may be outside the world border before taking
+     * damage.
+     *
+     * @return The distance
+     */
+    double safeZone();
+
+    /**
+     * Gets the damage done to a player per block per tick when outside the
+     * buffer.
+     *
+     * @return The damage amount
+     */
+    double damagePerBlock();
+
+    /**
+     * Gets the time when a contracting world border will warn a player for whom
+     * the world border will reach in {@code time} seconds.
+     *
+     * <p>In Minecraft, the warning is displayed in the form of a reddish
+     * tint.</p>
+     *
+     * @return The warning time
+     */
+    Duration warningTime();
```

Comment:
Isn't this more of ticks rather than duration?

Reply:
I was under the impression it was actually seconds - but I'll check that properly before it goes in.

Reply:
It is actually seconds, not ticks.